Kafka brokers certificates doesn't have the alternative name $serviceName.$namespace

(although i still need to test it with 0.7.0), when trying to connect from one NS with Kafka connect to another NS (cp-bkit-qb-kafka-cluster) running Kafka, I’m having the following TLS cert error

```
Caused by: java.security.cert.CertificateException: No subject alternative DNS name matching my-cluster-kafka-bootstrap.cp-bkit-qb-kafka-cluster found.
    at sun.security.util.HostnameChecker.matchDNS(HostnameChecker.java:214)
    at sun.security.util.HostnameChecker.match(HostnameChecker.java:96)
    at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:455)
    at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:436)
    at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:252)
    at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:136)
    at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1601)
    ... 16 more
```

Looks like the CERT created doesn't have the alternative name $serviceName.$namepsace

Can you confirm this is the case?  I would try this again with 0.7.0-rc1 just in case it changed

Regards

M

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java
@@ -1093,10 +1093,11 @@ public abstract class AbstractModel {
 
             Map<String, String> sbjAltNames = new HashMap<>();
             sbjAltNames.put("DNS.1", getServiceName());
-            sbjAltNames.put("DNS.2", String.format("%s.%s.svc.%s", getServiceName(), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
-            sbjAltNames.put("DNS.3", String.format("%s.%s.%s.svc.%s", podName.apply(cluster, i), getHeadlessServiceName(), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
+            sbjAltNames.put("DNS.2", String.format("%s.%s", getServiceName(), namespace));
+            sbjAltNames.put("DNS.3", String.format("%s.%s.svc.%s", getServiceName(), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
+            sbjAltNames.put("DNS.4", String.format("%s.%s.%s.svc.%s", podName.apply(cluster, i), getHeadlessServiceName(), namespace, KUBERNETES_SERVICE_DNS_DOMAIN));
 
-            int nextDnsId = 4;
+            int nextDnsId = 5;
 
             if (externalBootstrapAddress != null)   {
                 sbjAltNames.put("DNS." + nextDnsId, externalBootstrapAddress);

